econml/sklearn_extensions.py

```diff
@@ -0,0 +1,975 @@
+        self._X_offset = X_offset
+
+        # Calculate coefficient and error variance
+        num_nonzero_coefs = np.sum(self.coef_ != 0)
+        self._error_variance = np.average((y - y_pred)**2, weights=sample_weight) / \
+            (1 - num_nonzero_coefs / X.shape[0])
+        self._mean_error_variance = self._error_variance / X.shape[0]
+        self._coef_variance = self._get_unscaled_coef_var(
+            X, self._theta_hat, sample_weight) * self._error_variance
+
+        # Add coefficient correction
+        coef_correction = self._get_coef_correction(
+            X, y, y_pred, sample_weight, self._theta_hat)
+        self.coef_ += coef_correction
+
+        # Set coefficients and intercept standard errors
+        self.coef_std_err_ = np.sqrt(np.diag(self._coef_variance))
+        if self.fit_intercept:
+            self.intercept_std_err_ = np.sqrt(
+                np.matmul(np.matmul(self._X_offset, self._coef_variance), self._X_offset) +
+                self._mean_error_variance
+            )
+        else:
+            self.intercept_std_err_ = 0
+
+        # Set intercept
+        self._set_intercept(X_offset, y_offset, X_scale)
+        # Return alpha to 'auto' state
+        if self.selected_alpha_ is not None:
+            self.alpha = 'auto'
+        return self
+
+    def predict_interval(self, X, lower=5, upper=95):
+        """Build prediction confidence intervals using the debiased lasso.
+
+        Parameters
+        ----------
+        X : ndarray or scipy.sparse matrix, (n_samples, n_features)
+            Samples.
+
+        lower : float, optional
+            Lower percentile. Must be a number between 0 and 100.
+            Defaults to 5.0.
+
+        upper : float, optional
+            Upper percentile. Must be a number between 0 and 100, larger than 'lower'.
+            Defaults to 95.0.
+
+        Returns
+        -------
+        (y_lower, y_upper) : tuple of arrays, shape (n_samples, )
+            Returns lower and upper interval endpoints.
+        """
+        y_pred = self.predict(X)
+        y_lower = np.empty(y_pred.shape)
+        y_upper = np.empty(y_pred.shape)
+        # Note that in the case of no intercept, X_offset is 0
+        X = X - self._X_offset
+        # Calculate the variance of the predictions
+        var_pred = np.sum(np.matmul(X, self._coef_variance) * X, axis=1)
+        if self.fit_intercept:
+            var_pred += self._mean_error_variance
+
+        # Calculate prediction confidence intervals
+        sd_pred = np.sqrt(var_pred)
+        y_lower = y_pred + \
+            np.apply_along_axis(lambda s: norm.ppf(
+                lower / 100, scale=s), 0, sd_pred)
+        y_upper = y_pred + \
+            np.apply_along_axis(lambda s: norm.ppf(
+                upper / 100, scale=s), 0, sd_pred)
+        return y_lower, y_upper
+
+    def _get_coef_correction(self, X, y, y_pred, sample_weight, theta_hat):
+        # Assumes flattened y
```

Comment:
Do we have tests that assert that we do the right thing when inputs are of different shapes (not specifically for this method, but in general)?